src/commands/update.ts

```diff
@@ -0,0 +1,258 @@
+import chalk from 'chalk';
+import { tmpdir } from 'os';
+import pipe from 'promisepipe';
+import fetch from 'node-fetch';
+import Progress from 'progress';
+import { createGunzip } from 'zlib';
+import { parse, format } from 'url';
+import { basename, join } from 'path';
+import { spawnSync } from 'child_process';
+import {
+  createReadStream,
+  createWriteStream,
+  chmod,
+  chown,
+  copyFile,
+  move,
+  remove,
+  stat
+} from 'fs-extra';
+
+import pkg from '../util/pkg';
+import logo from '../util/output/logo';
+import handleError from '../util/handle-error';
+import getArgs from '../util/get-args';
+import { NowContext } from '../types';
+import createOutput, { Output } from '../util/output';
+
+const versionEndpoint = 'https://install-now-cli.zeit.sh/version';
+
+const platformMap: Map<string, string> = new Map([
+  ['darwin', 'macos'],
+  ['linux', 'linux'],
+  ['win32', 'win']
+]);
+
+const help = () => {
+  console.log(`
+  ${chalk.bold(`${logo} now update`)} [options]
+
+  ${chalk.dim('Options:')}
+
+    -h, --help                     Output usage information
+    -d, --debug                    Debug mode [off]
+    -c ${chalk.bold.underline('NAME')}, --channel=${chalk.bold.underline(
+    'NAME'
+  )}        Specify which release channel to install [stable]
+    -V ${chalk.bold.underline('VERSION')}, --version=${chalk.bold.underline(
+    'VERSION'
+  )}  Specfic version to install (overrides \`--channel\`)
+    -y, --yes                      Skip the confirmation prompt
+
+  ${chalk.dim('Examples:')}
+
+  ${chalk.gray('–')} Update Now CLI to the latest "canary" version
+
+      ${chalk.cyan(`$ now update --channel=canary`)}
+  `);
+};
+
+function detectAlpine() {
+  // https://github.com/sass/node-sass/issues/1589#issuecomment-265292579
+  const ldd = spawnSync('ldd', [process.execPath]).stdout.toString();
+  return /\bmusl\b/.test(ldd);
+}
+
+function getDefaultChannel(): string {
+  const m = pkg.version.match(/-(.+)\./);
+  return m ? m[1] : 'stable';
+}
+
+function getPlatform(nodePlatform: string): string {
+  let platform = platformMap.get(nodePlatform);
+  if (!platform) {
+    throw new Error(`Unsupported platform: ${platform}`);
+  }
+  if (platform === 'linux' && detectAlpine()) {
+    platform = 'alpine';
+  }
+  return platform;
+}
+
+function getReleaseUrl(version: string, platform: string): string {
+  const ext = platform === 'win' ? '.exe' : '';
```

Comment:
This was a little confusing. I thought it was a bug since platform is typically `win32` but I guess you already mapped it to a different naming convention 🤷‍♂

Reply:
Ya, `platform` is the Now CLI naming convention at this point.